Restructure and generate event data

juliehockett opened the pull request:
This will allow us to easily add additional test cases (by simply adding the JSON files and regenerating the Go file based on them). It also means that the event data will be embedded in the binary, so it can be run from anywhere.

CloudEvent samples are mostly nonexistent at this point, and the legacy ones are moved from the existing legacy data. We'll need to continue fleshing out these test cases.

juliehockett commented:
You can load the files from disk, you need only run `go generate ./...` prior to building and running the binary. Is that too much? So, the local workflow to add a test would be:

1. Add json input/output files to the data folder
2. Run `go generate ./...`
3. Run `go run github.com/functions-framework-conformance/client` with the appropriate commands

grant commented on client/main.go:
```diff
@@ -22,8 +22,9 @@ import (
-	functionType = flag.String("type", "http", "type of function to validate (must be 'http', 'cloudevent', or 'legacyevent'")
+	cmd             = flag.String("cmd", "", "command to run a Functions Framework server at localhost:8080")
+	functionType    = flag.String("type", "http", "type of function to validate (must be 'http', 'cloudevent', or 'legacyevent'")
+	validateMapping = flag.Bool("validate-mapping", true, "whether to validate mapping from legacy->cloud events and vice versa (as applicable)")
```
Is this flag documented?

juliehockett commented:
Added a readme to events/generate/data describing what a set of event inputs/outputs looks like and how to re-generate. Also added a travis check to ensure that the generation is included in the PR when changes are made.
